code/classes/DsaModelClass.py

```
@@ -201,19 +202,26 @@ def _clean_rgdp_pot(self):
         Clean baseline real potential growth.
         """
         for t, y in enumerate(range(self.start_year, self.end_year + 1)):
-            
-            # potential growth is based on OGWG up to T+5, long-run estimates from 2033, interpoalted in between
-            self.rg_pot_bl[t] = self.df_deterministic_data.loc[y, 'POTENTIAL_GDP_GROWTH']
-           
+            row = self.df_deterministic_data.loc[y]
+
+            # potential growth is based on OGWG up to T+5, long-run estimates from 2033, interpolated in between
+            rg = row.POTENTIAL_GDP_GROWTH
+            if pd.isna(rg):
+                raise MissingData("potential GDP growth", y, t)
+            self.rg_pot_bl[t] = rg
+
             # potential GDP up to T+5 are from OGWG, after that projected based on growth rate
-            self.rgdp_pot_bl[t] = self.df_deterministic_data.loc[y, 'POTENTIAL_GDP']
-            
-            # after T+5, potential GDP is projected based on growth rates form AWG
-            if pd.isna(self.df_deterministic_data.loc[y, 'POTENTIAL_GDP']):
-                self.rgdp_pot_bl[t] = self.rgdp_pot_bl[t - 1] * (1 + self.rg_pot_bl[t] / 100) 
+            rgdp = row.POTENTIAL_GDP
+            if pd.isna(rgdp) and t == 0:
+                raise MissingData("potential GDP", y, t)
```

If the very first data item is missing (`t == 0`), logically we cannot continue.

As it currently stands, this fails silently, because the code tries to forward project, analogous to the >= T+5 case. However, the "previous value" is read from `self.rgdp_pot_bl[-1]`, which is also NaN, so we still end up writing NaN.